general/mem_manager.cpp

```diff
@@ -1137,7 +1180,7 @@ void MemoryManager::Copy_(void *dst_h_ptr, const void *src_h_ptr,
       {
          if (dst_h_ptr != src_d_ptr && bytes != 0)
          {
-            internal::Memory &src_d_base = maps->memories.at(src_d_ptr);
+            internal::Memory &src_d_base = maps->memories.at(src_h_ptr);
```

Comment:
Should `src_d_base` be `src_h_base` ? Why are we changing only on one side ?

Reply:
The `internal::Memory` has both pointers, so both `*_d_*` and `*_h_*` in the name are not quite correct.

In any case, the bug was that the search key for `maps->memories` should be the host pointer, not the device.